bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,271 @@
+                if not is_spacing_valid(m.group("content"), control_spacing):
+                    yield self.handle_control_spacing_issue(
+                        file, filename, line, line_number, control_spacing, m)
+
+            # check for control end tags
+            for m in control_end_regex.finditer(line):
+                label = m.group("label")
+                try:
+                    (expected_label, start_in_line) = control_stack.pop()
+                except IndexError:
+                    # and end tag without an opening tag
+                    yield Result.from_values(
+                        origin=self,
+                        message="Control end tag has no corresponding start",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number + 1,
+                        end_column=m.end(0) + 1,
+                        severity=RESULT_SEVERITY.MAJOR)
+                    continue
+
+                # check for spacing issues with this tag
+                if not is_spacing_valid(m.group("content"), control_spacing):
+                    yield self.handle_control_spacing_issue(
+                        file, filename, line, line_number, control_spacing, m)
+
+                # yield results for incorrect or missing end labels
+                if label is None and line_number != start_in_line:
+                    diff = generate_label_diff(
+                        file, filename, line, line_number, m, expected_label)
+                    yield Result.from_values(
+                        origin=self,
+                        message="Untagged control end tag",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number,
+                        end_column=m.end(0) + 1,
+                        diffs=diff)
+                elif label != expected_label and line_number != start_in_line:
+                    diff = generate_label_diff(
+                        file, filename, line, line_number, m, expected_label)
+                    yield Result.from_values(
+                        origin=self,
+                        message="Tag does not match expected label",
+                        file=filename,
+                        line=line_number,
+                        column=m.start(0) + 1,
+                        end_line=line_number,
+                        end_column=m.end(0) + 1,
+                        diffs=diff)
+
+        # We've reached the end of the file.
+        # Check if all control blocks have been closed
+        if len(control_stack) > 0:
+            for (_, line_number) in control_stack:
+                yield Result.from_values(
+                    origin=self,
+                    message="Control start tag has no corresponding end",
```

Comment:
As we already have the label name, can we do "Control start tag 'tag_content' has no corresponding end tag" ? Does that make sense ?